android/libraries/rib-base/src/main/java/com/badoo/ribs/core/routing/configuration/ConfigurationContext.kt

```diff
@@ -0,0 +1,167 @@
+package com.badoo.ribs.core.routing.configuration
+
+import android.os.Bundle
+import android.os.Parcelable
+import com.badoo.ribs.core.Node
+import com.badoo.ribs.core.routing.action.RoutingAction
+import com.badoo.ribs.core.routing.configuration.ConfigurationContext.ActivationState.ACTIVE
+import kotlinx.android.parcel.Parcelize
+
+/**
+ * Describes a Configuration the [Node] can be in.
+ *
+ * [ConfigurationContext] can either be [ConfigurationContext.Unresolved], or
+ * [ConfigurationContext.Resolved]
+ */
+internal sealed class ConfigurationContext<C : Parcelable> {
+
+    abstract val configuration: C
+    abstract val activationState: ActivationState
+
+    /**
+     * Represents the activation state a [ConfigurationContext] can be in.
+     */
+    enum class ActivationState {
+        /**
+         * Off-screen configurations are considered [INACTIVE].
+         */
+        INACTIVE,
+
+        /**
+         * Logically active, but technically off-screen configurations (e.g. screen is not
```

Comment:
Probably I don't see bigger picture, but why should we even have this `SLEEPING` state?
If screen is not available during save/restore cycle, than it should be removed?
And the same question for `SleepingAction` etc. Can you explain?

Reply:
When the screen goes away, you would call `Deactivate` action on everything to execute `RoutingAction.cleanup()` and detach views. `Deactivate` by default also would mark all the elements as `INACTIVE`. Then after the screen comes back from save/restore cycle, everything is `INACTIVE`. How do you know which ones you need to reactivate? That's what the `SLEEPING` is for. It represents something that is not active now, but should be reactivated, whereas `INACTIVE` is intentionally not active and remains like that. Does it make more sense?

Reply:
Also we can discuss bigger picture in person.